src/scheduler.rs

```diff
@@ -168,6 +168,16 @@ async fn fetch_idle(ctx: &Context, connection: &mut Imap, folder: Config) -> Int
                 return connection.fake_idle(ctx, Some(watch_folder)).await;
             }
 
+            if folder == Config::ConfiguredInboxFolder {
+                if let Err(err) = connection
+                    .store_seen_flags(ctx)
```

Comment:
and here i'd prefer `set_seen_flags_on_imap` ... when using "save" or "store" it's nicer readable to disambiguate between imap/network things and pure local db changes.